src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  useState, useEffect, SetStateAction, Dispatch,
+} from 'react';
+
+export const useLocalStorage = <T>(
+  initialValue: T,
+  key: string,
+): [T, Dispatch<SetStateAction<T>>] => {
+  const getValue = (): T => {
+    const storageValue = localStorage.getItem(key);
+
+    if (storageValue) {
+      return JSON.parse(storageValue);
+    }
+
+    return initialValue;
+  };
+
+  const [value, setValue] = useState<T>(getValue());
+
+  useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(value));
```

Comment:
The best practice to use `try/catch` approach for any browser API (e.g. LocalStorage). It's not a mistake in case of this task but you can think about it for your future work.